Userland/Libraries/LibPDF/Parser.cpp

```diff
@@ -144,17 +145,17 @@ bool Parser::initialize_linearization_dict()
         || !number_of_pages.is_int_type<u16>()
         || !offset_of_main_xref_table.is_int_type<u32>()
         || (first_page && !first_page.is_int_type<u32>())) {
-        return true;
+        return false;
```

Comment:
Were this `true`s on purpose? I can't see why would they return `true` if `m_linearization_dictionary` hasn't been initialized. @mattco98

Reply:
Yes, they were. The return value of `false` indicated that the program should exit with an error, whereas `true` meant "yup, we're all good here" regardless of whether or not it was actually linearized (which is why the call to this function was immediately followed by `bool is_linearized = m_linearization_dictionary.has_value();`). So I don't think saying we always treated files as linearized is quite accurate, but regardless it's a pretty terrible design. Let's go with an enum return value here instead, something like:

```c++
enum class LinearizationResult {
    Error,
    NotLinearized,
    Linearized,
}
```

This is much closer to patterns elsewhere in the codebase, and makes a lot more sense :smile:

Reply:
Ahh, that makes more sense.